powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/annotation/PowerAuthAnnotationInterceptor.java

```diff
@@ -101,6 +112,17 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
                 }
             }
 
+            // Resolve @PowerAuthEncryption annotation
+            if (powerAuthEncryptionAnnotation != null) {
+                try {
+                    PowerAuthEciesEncryption eciesEncryption = this.encryptionProvider.validateEciesEncryption(request.getHeader(PowerAuthEncryptionHttpHeader.HEADER_NAME));
```

Comment:
This is what I am talking about - we should not be validating anything here, we should decrypt the actual request object and pass it so that it can be picked by the argument resolver...